core/src/main/java/org/elasticsearch/cluster/node/DiscoveryNode.java

```diff
@@ -238,6 +251,13 @@ public String getId() {
     }
 
     /**
+     * The unique ephemeral id of the node.
+     */
+    public String getEphemeralId() {
```

Comment:
see https://github.com/elastic/elasticsearch/pull/17987/files#r63505561
can we add some docs as to what we mean with ephemeral - most notably when it changes, and it's implication to identity?